src/commands/node.mjs

```diff
@@ -1040,28 +1009,347 @@ export class NodeCommand extends BaseCommand {
+
+          self.logger.debug('Initialized config', { config })
+        }
+      },
+      {
+        title: 'Identify existing network nodes',
+        task: async (ctx, task) => {
+          ctx.config.serviceMap = await self.accountManager.getNodeServiceMap(
+            ctx.config.namespace)
+          for (const serviceObject of ctx.config.serviceMap.values()) {
+            ctx.config.existingNodeIds.push(serviceObject.node)
+          }
+
+          return self.taskCheckNetworkNodePods(ctx, task, ctx.config.existingNodeIds)
+        }
+      },
+      {
+        title: 'Deploy new network node',
+        task: async (ctx, task) => {
+          const values = { hedera: { nodes: [] } }
+          let maxNum
+          for (const serviceObject of ctx.config.serviceMap.values()) {
+            values.hedera.nodes.push({
+              accountId: serviceObject.accountId,
+              name: serviceObject.node
+            })
+            maxNum = maxNum > AccountId.fromString(serviceObject.accountId).num ? maxNum : AccountId.fromString(serviceObject.accountId).num
+          }
+          for (const nodeId of ctx.config.nodeIds) {
+            const accountId = AccountId.fromString(values.hedera.nodes[0].accountId)
+            accountId.num = ++maxNum
+            values.hedera.nodes.push({
+              accountId: accountId.toString(),
+              name: nodeId
+            })
+          }
+
+          let valuesArg = ''
+          let index = 0
+          for (const node of values.hedera.nodes) {
+            valuesArg += `--set hedera.nodes[${index}].name=${node.name} --set hedera.nodes[${index}].accountId=${node.accountId} `
+            index++
+          }
+
+          await self.chartManager.upgrade(
+            ctx.config.namespace,
+            constants.FULLSTACK_DEPLOYMENT_CHART,
+            ctx.config.chartPath,
+            valuesArg
+          )
+          ctx.config.allNodeIds = [...ctx.config.existingNodeIds, ...ctx.config.nodeIds]
+        }
+      },
+      {
+        title: 'Check new network node pod is running',
+        task: async (ctx, task) => {
+          const subTasks = []
+          for (const nodeId of ctx.config.nodeIds) {
+            subTasks.push({
+              title: `Check new network pod: ${chalk.yellow(nodeId)}`,
+              task: async (ctx) => {
+                ctx.config.podNames[nodeId] = await this.checkNetworkNodePod(ctx.config.namespace, nodeId)
+              }
+            })
+          }
+
+          // setup the sub-tasks
+          return task.newListr(subTasks, {
+            concurrent: true,
+            rendererOptions: {
+              collapseSubtasks: false
+            }
+          })
+        }
+      },
+      {
+        title: 'Generate Gossip keys',
+        task: async (ctx, parentTask) => {
+          const config = ctx.config
+          const subTasks = self._nodeGossipKeysTaskList(config.keyFormat, config.allNodeIds, config.keysDir, config.curDate, false)
```

Comment:
Do we really need to generate keys for all nodes? 

We should only need to generate keys for the new node and send its public key to all other nodes.

Reply:
Good call.  I thought I added the logic, but on review my code was incomplete.  I've added the code and also added a test case to verify that the existing keys are maintained.